src/middlewared/middlewared/service.py

```diff
@@ -1097,15 +1104,17 @@ def reconfigure_logging(self):
         of the new location
         """
         reconfigure_logging()
-
-        # We need to kill this because multiprocessing has passed it stderr fd which is /var/log/middlewared.log
-        if osc.IS_LINUX:
-            for process in psutil.process_iter(attrs=["cmdline"]):
-                if "from multiprocessing.resource_tracker import main" in " ".join(process.info["cmdline"]):
-                    process.kill()
+        self.__kill_multiprocessing()
 
         self.middleware.send_event('core.reconfigure_logging', 'CHANGED')
 
+    @private
+    def stop_logging(self):
+        stop_logging()
+        self.__kill_multiprocessing()
+
+        self.middleware.send_event('core.reconfigure_logging', 'CHANGED', fields={'stop': True})
```

Comment:
Why don't you just make a `core.stop_logging` event?

Reply:
Thought about it but just make things harder for no reason.